oap-server/server-receiver-plugin/skywalking-telegraf-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/telegraf/provider/handler/TelegrafServiceHandler.java

```diff
@@ -0,0 +1,129 @@
+import org.apache.skywalking.oap.server.telemetry.api.HistogramMetrics;
+import org.apache.skywalking.oap.server.telemetry.api.MetricsCreator;
+import org.apache.skywalking.oap.server.telemetry.api.MetricsTag;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@Slf4j
+public class TelegrafServiceHandler {
+
+    private final HistogramMetrics histogram;
+    private final CounterMetrics errorCounter;
+    private List<MetricConvert> metricConvert;
+
+    public TelegrafServiceHandler(ModuleManager moduleManager, MeterSystem meterSystem, List<Rule> rules) {
+
+        this.metricConvert = rules.stream().map(r -> new MetricConvert(r, meterSystem)).collect(Collectors.toList());
+
+        final MetricsCreator metricsCreator = moduleManager.find(TelemetryModule.NAME)
+                                                           .provider()
+                                                           .getService(MetricsCreator.class);
+
+        histogram = metricsCreator.createHistogramMetric(
+                "telegraf_in_latency", "The process latency of telegraf data",
+                new MetricsTag.Keys("protocol"), new MetricsTag.Values("http")
+        );
+
+        errorCounter = metricsCreator.createCounter(
+                "telegraf_error_count", "The error number of telegraf analysis",
+                new MetricsTag.Keys("protocol"), new MetricsTag.Values("http")
+        );
+    }
+
+    /**
+    * Convert the TelegrafData object to meter {@link Sample}
+    **/
+    public List<Sample> convertTelegraf(TelegrafDatum telegrafData) {
+
+        List<Sample> sampleList = new ArrayList<>();
+
+        Map<String, Object> fields = telegrafData.getFields();
+        String name = telegrafData.getName();
+        Map<String, String> tags = telegrafData.getTags();
+        ImmutableMap<String, String> immutableTags = ImmutableMap.copyOf(tags);
+        long timestamp = telegrafData.getTimestamp();
+
+        fields.forEach((key, value) -> {
+            if (value instanceof Number) {
+                Sample.SampleBuilder builder = Sample.builder();
+                Sample sample = builder.name(name + "_" + key)
+                        .timestamp(timestamp * 1000L)
+                        .value(((Number) value).doubleValue())
+                        .labels(immutableTags).build();
+
+                sampleList.add(sample);
+            }
+        });
+        return sampleList;
+
+    }
+
+    public ImmutableMap<String, SampleFamily> convertSampleFamily(TelegrafData telegrafData) {
+        List<Sample> allSamples = new ArrayList<>();
+
+        List<TelegrafDatum> metrics = telegrafData.getMetrics();
+        for (TelegrafDatum m : metrics) {
+            List<Sample> samples = convertTelegraf(m);
+            allSamples.addAll(samples);
+        }
+
+        // Grouping all samples by their name, then build sampleFamily
```

Comment:
This policy seems not right. The samples should be grouped by bulk/sampling ID. 
Because 
1. Metrics in the same name with different timestamps should belong to different sample families.
2. Different metrics(name) collected in the same timestamp(one collecting) should belong to the same sample family. You could check the MAL doc, which mentioned metrics should be calculated through one formula, ref to https://skywalking.apache.org/docs/main/next/en/concepts-and-designs/mal/#binary-operators, `instance_trace_analysis_error_count / instance_trace_count`

Reply:
> This policy seems not right. The samples should be grouped by bulk/sampling ID. Because
> 
> 1. Metrics in the same name with different timestamps should belong to different sample families.
> 2. Different metrics(name) collected in the same timestamp(one collecting) should belong to the same sample family. You could check the MAL doc, which mentioned metrics should be calculated through one formula, ref to https://skywalking.apache.org/docs/main/next/en/concepts-and-designs/mal/#binary-operators, `instance_trace_analysis_error_count / instance_trace_count`

Got it. I will modify this part logic.

Reply:
> This policy seems not right. The samples should be grouped by bulk/sampling ID. Because
> 
> 1. Metrics in the same name with different timestamps should belong to different sample families.
> 2. Different metrics(name) collected in the same timestamp(one collecting) should belong to the same sample family. You could check the MAL doc, which mentioned metrics should be calculated through one formula, ref to https://skywalking.apache.org/docs/main/next/en/concepts-and-designs/mal/#binary-operators, `instance_trace_analysis_error_count / instance_trace_count`

I already fix the issue by grouping sample family by name and timestamp, and extra UTs have been tested.